js/FileManager.js

```diff
@@ -0,0 +1,156 @@
+	var current_extension=filename.substring(extension_position,filename.length);
+	if(current_extension.toLowerCase()!=extension.toLowerCase()) {
+		filename+=extension;
+	}
+	return filename;
+}
+
+// Set the title to include the model name
+function set_title(filename) {
+	var title;
+	if(filename!=null) {
+		title = filename+"|InsightMaker";
+		
+	} else {
+		title = "InsightMaker";
+	}
+	window.parent.document.title = title;
+}
+
+// Get xml data for the current model
+function get_model_xml() {
+	var enc = new mxCodec();
+	var graph_dom=enc.encode(graph.getModel());
+	var xml_data="<InsightMakerModel>"+graph_dom.innerHTML+"</InsightMakerModel>";
+	return xml_data;
+}
+
+// Makes a new model
+function newModel() {
+	clearModel();
+}
+
+// Class for uploading and downloading files from javascript
+var WebFileIO = new function() {
+	var upload_handler;
+	function readfile(event) {
+		var file = event.target.files[0]; 
+
+		if (file) {
+		  var reader = new FileReader();
+		  reader.onload = function(reader_event) { 
+			  var filedata = reader_event.target.result;
+			  upload_handler(file.name, filedata);	
+		  }
+		  reader.readAsText(file);
+		}
+	}
+
+	this.upload = function(tupload_handler, accepttype) {
+		var uploader = document.body.appendChild(document.createElement("input"));
+		uploader.addEventListener('change', readfile, false);
+		uploader.type="file";
+		if(accepttype!==undefined) {
+			// accepttype is file extension that we should show
+			// E.g. ".txt"
+			// If none is specified all file extensions are shown
+			uploader.accept=accepttype;
+		}
+		upload_handler = tupload_handler;
+		
+		uploader.click();
+		uploader.parentElement.removeChild(uploader);
+	};
+	
+	this.download = function(filename, content) {
+		var downloadlink = document.body.appendChild(document.createElement("a"));
+		downloadlink.download = filename;
+		
+		downloadlink.href = "data:text/plain;base64," + btoa(content);
+		downloadlink.click();
+		downloadlink.parentElement.removeChild(downloadlink);
+	};
+}
+
+// High-level File manager. Does save and load of models
+var FileManagerWeb = new function() {
+	var self = this;
+	var filename = null;
+	
+	this.set_filename = function(filename) {
+		self.filename=filename;
+		set_title(filename);
+	}
+	
+	this.saveModel = function() {
+		var xml_data = get_model_xml();
+		
+
+
+		var model_name=prompt("Enter name of model");
+		if(model_name==null) {
+			return;
+		}
+		model_name=append_file_extension(model_name,InsightMakerFileExtension);
+		self.set_filename(model_name);
+		WebFileIO.download(model_name,xml_data);
+	};
+	
+	this.loadModel = function() {
+		WebFileIO.upload(function(filename,model_data) {
+		importMXGraph(model_data);
+		self.set_filename(filename);
+		},InsightMakerFileExtension);
+	};
+	
+	this.newModel = function() {
+		self.set_filename(null);
+		newModel();
+	}
+};
+
+// FileMenu for environment.WebOffline
+var FileMenuWeb = {
+text: getText('File'),
+itemId: "filegroup",
```

Comment:
When we're showing the file group, should we hide the Share -> Import/Export menu items? It seems there is overlap in functionality.